firebase-firestore/src/main/java/com/google/firebase/firestore/local/LruGarbageCollector.java

```diff
@@ -99,6 +107,43 @@ public int getDocumentsRemoved() {
     }
   }
 
+  public class Scheduler {
```

Comment:
I like this PR! While there is some special-casing in FirestoreClient with this PR, I don't think it's necessarily more than before, and some of it might even go away if we always run LRU (for Memory persistence for example). The encapsulation in `Scheduler` is really nice and hides all implementation details.  Thanks.